### DISCUSSION TITLE
Cannot update email for user account

### DISCUSSION THREAD

**Discussion by brzozowskia** [dani-garcia/vaultwarden#3760] [Q&A]
Unable to update account email on vaultwarden. 

Vaultwarden requires an email verification step. However, the verification email is never received. Without going through this process, the user account email cannot be changed.

Using the most current version of vaultwarden as of this post.

**Comment by stefan0xC** [comment]
Can you provide more information about your setup? Do you have SMTP enabled? What do the logs say when vaultwarden tries to send a mail?

**Comment by BlackDex** [comment]
Best is to provide the diagnostics string which you can generate via the /admin/diagnostics

**Comment by BlackDex** [comment]
I tested it, and it works just fine, as long as SMTP is enabled.
The only thing we could try to do is report an error if SMTP is disabled.

The flow of Bitwarden regarding changing the mail address mandates a working SMTP connection.

**Comment by DanielAndreasen** [comment]
I had the same issue, and was not ready to set up SMTP. Here is a hacky way, that should be used with caution.

1) I opened the database for my vaultwarden instance. For me it is a sqlite database, so simple do `sqlite3 db.sqlite` did the trick.
2) Get the token after requesting to change the email from the database; `select email, email_new, email_new_token from users;`. Confirm your old email is listed next to the new you requested. Then enter the token in the form online.

I did not want to update the email directly in the database, because I don't know if there are other dependencies.

**Reply by spyderdan217** [reply]
Thanks for this. I ran into the same issue where for some reason I did not receive my verification code to change email address. All other SMTP functions work fine however.

I used this trick to finish my process
